mp/src/game/server/momentum/mom_timer.cpp

```diff
@@ -398,6 +398,13 @@ CON_COMMAND_F(mom_restart_stage,
         track = pPlayer->m_Data.m_iCurrentTrack;
     }
 
+    if (pPlayer->m_iZoneCount[track] == 1)
```

Comment:
I think the better logic to use here is the linear tracks array inside the player.